src/network.zig

```diff
@@ -851,6 +851,11 @@ pub const Protocols = struct {
 		fn receive(conn: *Connection, _: []const u8) !void {
 			conn.disconnect();
 			if(conn.user) |user| {
+				main.server.world.?.savePlayer(user) catch |err| {
```

Comment:
This is not the right place.

The update thread might still use the player and collect items into their inventory while the disconnect happens.

Instead you should do this in the user.deinit() function.